routemaster/state_machine/tests/test_actions.py

```diff
@@ -62,6 +66,62 @@ def test_actions_do_not_advance_state_on_fail(app_config, create_label, mock_web
     ])
 
 
+def test_actions_retries_use_same_idempotency_token(app_config, create_label, mock_webhook, assert_history):
+    state_machine = app_config.config.state_machines['test_machine']
+
+    # Figure out what our idempotency token should be
```

Comment:
this couples the test to the implementation; I'd rather have a test which learns what it should be from the first action and then ensures that the second call matches the first.

Reply:
Yeah I wanted to do that but it looked quite tricky. I'll see what I can do.

Reply:
fe9dc27608b651d92912681589bf12f6744a6668

Reply:
Ah, in the past I've used [`call_args`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args) or [`call_args_list`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args_list) for getting the arguments from the first call. Unfortunately that ends up being about as complex as you need to cope with them being empty in the failure cases.

Reply:
for clarity: no need to change this